scripts/importer/data_importer/raw_data_importer.py

```diff
@@ -0,0 +1,502 @@
+#!/usr/bin/env python3
+import re
+import sys
+import json
+import time
+import logging
+
+from datetime import datetime
+
+import modules.browser.lookups
+import db
+from .data_importer import DataImporter
+
+METRICS = [
+    'BaseQRankSum',
+    'ClippingRankSum',
+    'DP',
+    'FS',
+    'InbreedingCoeff',
+    'MQ',
+    'MQRankSum',
+    'QD',
+    'ReadPosRankSum',
+    'VQSLOD'
+]
+
+def get_minimal_representation(pos, ref, alt):
+    """
+    Get the minimal representation of a variant, based on the ref + alt alleles in a VCF
+    This is used to make sure that multiallelic variants in different datasets,
+    with different combinations of alternate alleles, can always be matched directly.
+
+    Note that chromosome is ignored here - in xbrowse, we'll probably be dealing with 1D coordinates
+    Args:
+        pos (int): genomic position in a chromosome (1-based)
+        ref (str): ref allele string
+        alt (str): alt allele string
+    Returns:
+        tuple: (pos, ref, alt) of remapped coordinate
+    """
+    pos = int(pos)
+    # If it's a simple SNV, don't remap anything
+    if len(ref) == 1 and len(alt) == 1:
+        return pos, ref, alt
+    else:
+        # strip off identical suffixes
+        while(alt[-1] == ref[-1] and min(len(alt), len(ref)) > 1):
+            alt = alt[:-1]
+            ref = ref[:-1]
+        # strip off identical prefixes and increment position
+        while(alt[0] == ref[0] and min(len(alt), len(ref)) > 1):
+            alt = alt[1:]
+            ref = ref[1:]
+            pos += 1
+        return pos, ref, alt
+
+
+class RawDataImporter(DataImporter):
+    def __init__(self, settings):
+        super().__init__(settings)
+        self.dataset_version = None
+        self.dataset = None
+        self.sampleset = None
+        self.counter = {'coverage':None,
+                        'variants':None}
+
+    def _set_dataset_info(self):
+        """ Save dataset information given as parameters """
+        if self.settings.beacon_description:
+            self.dataset.description = self.settings.beacon_description
+            self.dataset.save()
+        if self.settings.assembly_id:
+            self.dataset_version.var_call_ref = self.settings.assembly_id
+            self.dataset_version.save()
+        if self.settings.sampleset_size:
+            self.sampleset.sample_size = self.settings.sampleset_size
+            self.sampleset.save()
+        if self.settings.dataset_size:
+            self.dataset.dataset_size = self.settings.dataset_size
+            self.dataset.save()
+
+    def _select_dataset_version(self):
```

Comment:
This was an extremely big subroutine for selecting the correct dataset version. I would have just forced the user to specify the exact name of the version. I'm not the one that has to run this so maybe it's really convenient for the person doing the imports, otherwise I would instead just implement a small CLI that can list all versions and datasets in the database (and create new ones).

Reply:
Seems most of it is removed and version must now be defined as a parameter